modules/ogc/net.opengis.wfs/src/net/opengis/wfs20/impl/SimpleFeatureCollectionTypeImpl.java

```diff
@@ -178,7 +178,7 @@ public Object eGet(int featureID, boolean resolve, boolean coreType) {
     public void eSet(int featureID, Object newValue) {
     switch (featureID) {
       case Wfs20Package.SIMPLE_FEATURE_COLLECTION_TYPE__BOUNDED_BY:
-        setBoundedBy((EnvelopePropertyType)newValue);
+        setBoundedBy(newValue instanceof EnvelopePropertyType ? (EnvelopePropertyType) newValue : null);
```

Comment:
Q: What is it getting back that is not an EnvelopePropertyType?

Reply:
I see, it is explicitly allowed to be null:

```
<complexType name="BoundingShapeType">
  <sequence>
    <choice>
      <element ref="gml:Envelope"/>
      <element ref="gml:Null"/>
    </choice>
  </sequence>
  <attribute name="nilReason" type="gml:NilReasonType"/>
</complexType>
```

Reply:
> 
> 
> Q: What is it getting back that is not an EnvelopePropertyType?

Hi Jody, it get an empty String, because the parsing failed.

Reply:
> 
> 
> I see, it is explicitly allowed to be null:
> 
> ```
> <complexType name="BoundingShapeType">
>   <sequence>
>     <choice>
>       <element ref="gml:Envelope"/>
>       <element ref="gml:Null"/>
>     </choice>
>   </sequence>
>   <attribute name="nilReason" type="gml:NilReasonType"/>
> </complexType>
> ```

When I look into the actual schema (http://schemas.opengis.net/wfs/2.0/wfs.xsd) I see the following:

![grafik](https://user-images.githubusercontent.com/11347841/91268715-1a341700-e776-11ea-9eb2-77b076744029.png)

Are GeoTools using this schema?

I think the problem is the "any" type...

Reply:
GeoTools is using that schema. Being a schema driven parser, it might well be that its having problems when dealing with a "any" target, it probably does not know what bindings to use. 
@awaterme if you're interested, try adding a breakpoint in XSAnyTypeBinding and see if that's the one it is using to parse stuff.
@nmco and @fernandor777 deal with this code more than I do, maybe they have a more educated guess as to what is happening.